src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,138 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    List mylist = new ArrayList();
+
+    public static void main(String[] args) {
+//        Range a = Range.of(1, 10);
+//        a.test();
+//        System.out.println("size "+a.size());
+//        System.out.println("is empty?" + a.isEmpty());
+//        System.out.println(a.add(0.2f));
+//        System.out.println(a.contains(5));
+//        a.test();
+    }
+
+    public Range(float a, float b) {
+        if(!(a==b)){
+            while (a <=b) {
+                mylist.add(a);
+                a = a + 0.1f;
+                a = a * 10;
+                a = Math.round(a);
+                a = a / 10;
+
+            }}
+    }
+
+    public Range(int a, int b) {
+        if (!(a==b)){
+            while (a <= b) {
+                mylist.add(a);
+                a = a + 1;
+            }}
+
+    }
+
+    public static Range of(int i, int i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range of(float i, float i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range of(char i, char i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> characterCharacterFunction) {
+        return null;
+    }
+
     public int size() {
-        return 0;
+        return mylist.size();
     }
 
     public boolean isEmpty() {
-        return false;
+        return mylist.isEmpty();
     }
 
+    @Override
     public boolean contains(Object o) {
-        return false;
+        return mylist.contains(o);
     }
 
-    public Iterator<T> iterator() {
-        return null;
+    @Override
+    public Iterator iterator() {
+        return mylist.iterator();
     }
 
+    @Override
     public Object[] toArray() {
         return new Object[0];
     }
 
-    public <T1> T1[] toArray(T1[] a) {
-        return null;
+    @Override
+    public boolean add(Object o) {
```

Comment:
Set doesn't contain duplications, but here looks like if you add two equal values both of them will be saved. The same for method addAll

Reply:
corrected